hack/update-codegen.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cd "$(dirname "${0}")"
+GO_CMD=${1:-go}
+
+"${GO_CMD}" mod download
+CODEGEN_PKG="$(go list -m -f '{{.Dir}}' k8s.io/code-generator)"
+
+bash "${CODEGEN_PKG}/generate-groups.sh" \
+  "all" \
+  sigs.k8s.io/kueue/pkg/client \
```

Comment:
```suggestion
  sigs.k8s.io/kueue/apis/client \
```

Perhaps this is better?

Reply:
In other sigs projects, we used to put the client on the `pkg/client` or `pkg/generated`.

https://github.com/search?q=org%3Akubernetes-sigs+generate-groups.sh&type=code

Do you prefer `apis/client `?

Reply:
It doesn't look very consistent.

I want it to be more similar to k/k (`k8s.io/api` and `k8s.io/client-go`) . Then maybe it should be `sigs.k8s.io/client-go`

Reply:
> I want it to be more similar to k/k (k8s.io/api and k8s.io/client-go) 

I see. Let's move generated codes to `apis/client`

Honestly, I don't have a particular opinion about that :)

Reply:
Let's do `sigs.k8s.io/client-go` (without `apis`)

Reply:
Sure!